common/lib/xmodule/xmodule/html_module.py

```diff
@@ -117,9 +119,7 @@ def student_view_data(self, context=None):  # pylint: disable=unused-argument
     def get_html(self):
         """ Returns html required for rendering the block. """
         # pylint: disable=no-member
-        if self.data is not None and getattr(self.system, 'anonymous_student_id', None) is not None:
-            return self.data.replace("%%USER_ID%%", self.system.anonymous_student_id)
-        return self.data
+        return substitute_keywords_with_data(self.data, self.system)
```

Comment:
Won't this also add several other new %%field%% substitutions, like `%%COURSE_END_DATE%%` ?

I'd prefer a simple change like the following. This is (AFAIK) the most current way to write this code, not using old modulesystem properties that I consider deprecated:
```
if self.data is not None:
    data = self.data
    data = data.replace("%%USER_ID%%", getattr(self.runtime, 'anonymous_student_id', "%%USER_ID%%"))
    data = data.replace("%%COURSE_ID%%", block.scope_ids.usage_id.context_key)
    return data
```

This also avoids reaching into `common.djangoapps.util` (built-in XBlocks should try to reduce dependencies on other parts of edx-platform IMHO)

Reply:
This was my first approach, but we ended proposing the alternative one. I changed this according to your review. Thank you!